classes/local/store/digitalocean/client.php

```diff
@@ -53,10 +53,32 @@ public function set_client($config) {
         ));
     }
 
-    public function define_client_section($mform, $config) {
+    public function define_client_check($client) {
```

Comment:
this seems duplicated from the base class `object_client_base`

Reply:
`define_client_check` overrides `object_client_base`'s function as it needs to supply false to the test permissions function

Reply:
you could set a property inside `object_client_base` with default value of `true` then override this value in `digitalocean/client` and set it to false.
Then you could:
```
// Check permissions if we can connect.
$permissions = $client->test_permissions($this->testdelete);
```
Also for consistency it would be a good idea to update the signature of `test_permissions` for the azure/swift clients, perhaps it should be included in the `interface`?